### ISSUE TITLE
GameGuru MAX - Snapping on the Up Axis - Request

### ISSUE THREAD

**Issue by MonkeyFrogStudio** [Dark-Basic-Software-Limited/GameGuruRepo#1568]
Snapping in MAX currently only occurs on the two horizontal axis, excepting snapping to the tops of entities in the scene. I would like to also see snapping along the vertical axis as well, especially when moving entities about using the gizmo/widget. This is pretty standard every 3D modeling package and every 3D game engine I've ever used.

**Comment by Teabone3** [comment]
Yes please.
I've always had troubles getting things properly aligned in both GGC and GGMax. GGMax has the awesome ability now to put things exactly where you want them by number inputs -- which is a fantastic feature. Though I think for snapping we need some assistance with height placement so its not as tedious to place. 

So i for one second this suggestion for sure. I had spent an 30 minutes the other day making a single room, which would take me mere minutes (2 to 5 minutes) in most other engines (even FPSC).

**Comment by LeeBamberTGC** [comment]
@MonkeyFrogStudio Yes, I think extending the XZ system to include Y would be the way to go, and the widget is certainly the most 3D modeller-friendly way. Do you see how it would work with the non-widget system?  Maybe it kicks in when you use the vertical positioning mode?

**Comment by MonkeyFrogStudio** [comment]
@LeeBamberTGC I definitely think, if snapping is turned on, that it should kick in when vertical positioning is used. 

In 3D modeling software, "up" is determined, when not using the widget, by your view position. So, if your view is **mostly** top-down and you grab-move an object without a widget, then it would move along the X and Y axis only. If your view is mostly straight on, similar to a first-person view, then dragging the object would be restricted to snapping (and moving) along the XY or ZY (depending on the view). So, just dragging on object without the widget would not allow you to move along all three axis, but only two of them at a time. Which two is determined by the view angle.

**Comment by MonkeyFrogStudio** [comment]
@LeeBamberTGC I also see snapping on the Y axis affecting PageUp/PageDown. So, the end-user could set a Y snapping, tap PageUp and whatever object is selected would snap UP that amount. Set Y to 32 and pressing PageUp would snap the object upward to the next 32 snap grid point. If it was at zero, then it would move up to 32. If it was at 32, it would move up to 64. If the object were at 10 on the Y, then it would move the object up to exactly 32 (snapping to the nearest "grid", etc.).

**Comment by MonkeyFrogStudio** [comment]
@LeeBamberTGC Also, while we're at it, a few more considerations:

Rotation snapping for the widget:
https://github.com/TheGameCreators/GameGuruRepo/issues/3458

Since I'd mentioned PageUp/PageDown above:
https://github.com/TheGameCreators/GameGuruRepo/issues/3022

Why have two fields for snapping (X and Z) when most only have one snapping setting that affects all axis (at least add a "maintain aspect ratio" checkbox where changing one causes the others to change equal to it):
https://github.com/TheGameCreators/GameGuruRepo/issues/2667

And simply because I really, really want this to happen:
https://github.com/TheGameCreators/GameGuruRepo/issues/2427

The above link is about having MAX remember our settings on exit so we don't have to keep resetting them each time we open MAX. I test a lot of scripts for Necrym59 and that means I am constantly exiting and restarting MAX. I also make a lot of changes as I test the parts/objects I make for use in MAX. Again, this often means exiting and restarting MAX. And each time I have to go to grid snapping and reset it to what I was using when I exited. If MAX would simply REMEMBER my settings, I could just open MAX and dive straight into developing. It's a bit of a PITA to have to keep changing these settings each time. :)

**Comment by MonkeyFrogStudio** [comment]
Snapping on the Y axis has been added, so I'm closing this. :)
